CMakeLists.txt

```diff
@@ -14,9 +14,16 @@ endif()
 add_library(mbed-os OBJECT)
 
 include(${MBED_CONFIG_PATH}/mbed_config.cmake)
+
 include(${MBED_ROOT}/cmake/core.cmake)
+mbed_set_cpu_core_options(mbed-os ${MBED_TOOLCHAIN})
```

Comment:
can you illustrate the problem why we need all these `mbed_set_`  ?

My understanding is: every component for Mbed Os would need to call these to set expected compiler/linker flags? wouldn't it be easier if we do it all in one call rather or that is not feasible ?
If I add new component into Mbed OS tree, lets say new-driver, would this component include in its CMakelists all these `mbed_set` ? or just some or does not need any?

Reply:
My understanding is that we want to be able to break down the `mbed-os` target into a number of targets that will be linked to the `mbed-os` target depending on user preferences. Each target need therefore to be built on its own and therefore should be able to set the toolchain options it needs to do so.

We made a decision to use `target_compile_options()` and `target_link_options()` rather than using global variables that apply the options to all targets built from a given top level CMake source file. These functions require a target to be specified to use them.

Say we build `mbed-os` and `target_link_libraries()` with the (to be written) `rtos` or `events` targets, each of these two targets need to be built separately from the `mbed-os` target when we do `add_subdirectory(rtos/source/TARGET_CORTEX)` and `add_subdirectory(events)`. They will need to set `profile`, `core` and `toochain` compile and link options. They will therefore need to pass their respective target names to the added functions.

Reply:
My understanding is that we want to be able to break down the `mbed-os` target into a number of targets that will be linked to the `mbed-os` target depending on user preferences. Each target need therefore to be built on its own and therefore should be able to set the toolchain options it needs to do so.

We made a decision to use `target_compile_options()` and `target_link_options()` rather than using global variables that apply the options to all targets built from a given top level CMake source file. These functions require a target to be specified to use them.

Say we build `mbed-os` and `target_link_libraries()` with the (to be written) `rtos` or `events` targets, each of these two targets need to be built separately from the `mbed-os` target when we do `add_subdirectory(rtos/source/TARGET_CORTEX)` and `add_subdirectory(events)`. They will need to set `profile`, `core` and `toochain` compile and link options. They will therefore need to pass their respective target names to the added functions.

Reply:
What constitute the `mbed-os` target is what we include when we build the bare metal profile, so `platform`, `drivers`, `rtos-api`, as well as the stuff under `cmsis`, `hal`, and `targets`. These directories will not be separate CMake targets as they will be integral to Mbed OS (the minimum requirement to use the OS).

Everything else (identified by the presence of `mbed_lib.json` in the dir) should be, IMO, an optional independent add on library (therefore CMake target) that can be added if the users wants it.

Reply:
Adding this functions we provide default flags components can use.

Reply:
> Adding this functions we provide default flags components can use.

Yes, that was the idea.